drivers/flash/nrf_qspi_nor.c

```diff
@@ -502,11 +502,16 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 		return -EINVAL;
 	}
 
+	u8_t buffer[size] __attribute__ ((aligned (32)));
```

Comment:
This isn't going to work, because it could overrun the stack.  It's also very inefficient.

Reply:
Thank you @pabigot.
I'm very new to low level programming - are you able to explain why this could cause a stack overrun?

Reply:
You're dynamically allocating a buffer of `size` bytes on the stack to hold the aligned copy.  Since `size` is an input to the function there's no bound on how large it is: the caller has guaranteed to provide that much data, but the function can't affort to make a copy of it.